syntax = "proto3";
// python3 -m grpc_tools.protoc -I=. --python_out=. --pyi_out=. --grpc_python_out=. mempool.proto
package mempool;

service MemPool {
  rpc Add(UserOp) returns (AddResponse) {}
  rpc Get(GetRequest) returns (GetResponse) {}
}

message UserOp {
  bytes sender = 1;
  bytes nonce = 2;
  bytes init_code = 3;
  bytes call_data = 4;
  bytes call_gas_limit = 5;
  bytes verification_gas_limit = 6;
  bytes pre_verification_gas = 7;
  bytes max_fee_per_gas = 8;
  bytes max_priority_fee_per_gas = 9;
  bytes paymaster_and_data = 10;
  bytes signature = 11;
}

message AddResponse {
  uint64 userop_id = 1;
}

message GetRequest {
  string from = 1;
  string to = 2;
}

message GetResponse {
  repeated UserOp userOps = 1;
}